### PR TITLE
Attempted Mini Challenge 4

### PR DESCRIPTION
**Pull request by goodnessemmanuel**
Challenges attempted with out recent new test cases

### CODE REVIEW

File: src/word-search/index.js [hunk 1]
```diff
@@ -1,3 +1,59 @@
...
+            {
+                holdWord = [];//refresh the container at the start of each new search transverse 
+                searchMechanism(words, i, j, word, indexToSearch, searchTrack, holdWord);
+            }
+        }
+    }
+    //if you find the word return true, if not, return false.
+    return holdWord.join('') === word? true : false;
+}
+
```
Comment (flinchy) [inline src/word-search/index.js:58]: Update this branch with the master branch, and make sure all your test cases are passing, new test cases were added

File: src/count-islands/index.js [hunk 2]
```diff
@@ -1,3 +1,64 @@
...
+function countIslands(grid) 
+{
+    //check to ensure a valid array was entered
+    if (!Array.isArray(grid))
+    {
+        return;
+    }
+
+    //function variable to keep track of neigbouring lands(1's) with respect to the current land(1)
+    let searchIslands = function (islandArray, row, col, track)
```
Comment (flinchy) [inline src/count-islands/index.js:10]: move searchIsLands() function out of countIsLands()

File: src/count-islands/index.js [hunk 3]
```diff
@@ -1,3 +1,64 @@
-function countIslands(grid) {}
+function countIslands(grid)
```
Comment (flinchy) [inline src/count-islands/index.js:1]: fix your failing test cases

File: src/word-search/index.js [hunk 4]
```diff
@@ -1,3 +1,59 @@
-function wordSearch(words, word) {}
+function wordSearch(words, word)
```
Comment (flinchy) [inline src/word-search/index.js:1]: fix your failing test cases

Comment (goodnessemmanuel) [reply src/word-search/index.js:1]: Sir, I have updated my branch with the new test cases.

**Comment by flinchy** [comment]
cool

**Pull request merged**